OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -741,31 +742,33 @@ bool SwitchTeamChat()
 		void UpdateCurrentMap()
 		{
 			var uid = orderManager.LobbyInfo.GlobalSettings.Map;
-			if (Map.Uid == uid)
+			if (MapPreview.Uid == uid)
 				return;
 
-			Map = Game.ModData.MapCache[uid];
-			if (Map.Status == MapStatus.Available)
+			MapPreview = Game.ModData.MapCache[uid];
+			if (MapPreview.Status == MapStatus.Available)
 			{
 				// Maps need to be validated and pre-loaded before they can be accessed
 				new Thread(_ =>
 				{
-					var map = Map;
-					map.CacheRules();
+					Map = new Map(MapPreview.Path);
+					Map.PreloadRules();
```

Comment:
If this thread and the map caching thread happen to be loading the same map at the same time this will likely explode - they'll be seeking around in the same streams when loading the files from the mod container and messing with each other's offsets - super bad times.